lib/configs/flat/vue2-essential.js

```diff
@@ -5,72 +5,74 @@
-  'vue/no-multiple-template-root': 'error',
-  'vue/no-mutating-props': 'error',
-  'vue/no-parsing-error': 'error',
-  'vue/no-ref-as-operand': 'error',
-  'vue/no-reserved-component-names': 'error',
-  'vue/no-reserved-keys': 'error',
-  'vue/no-reserved-props': [
-    'error',
-    {
-      vueVersion: 2
+module.exports = [
+  ...config,
+  {
+    rules: {
+      'vue/multi-word-component-names': 'error',
+      'vue/no-arrow-functions-in-watch': 'error',
+      'vue/no-async-in-computed-properties': 'error',
+      'vue/no-child-content': 'error',
+      'vue/no-computed-properties-in-data': 'error',
+      'vue/no-custom-modifiers-on-v-model': 'error',
+      'vue/no-dupe-keys': 'error',
+      'vue/no-dupe-v-else-if': 'error',
+      'vue/no-duplicate-attributes': 'error',
+      'vue/no-export-in-script-setup': 'error',
+      'vue/no-multiple-template-root': 'error',
+      'vue/no-mutating-props': 'error',
+      'vue/no-parsing-error': 'error',
+      'vue/no-ref-as-operand': 'error',
+      'vue/no-reserved-component-names': 'error',
+      'vue/no-reserved-keys': 'error',
+      'vue/no-reserved-props': [
+        'error',
+        {
+          vueVersion: 2
+        }
+      ],
+      'vue/no-shared-component-data': 'error',
+      'vue/no-side-effects-in-computed-properties': 'error',
+      'vue/no-template-key': 'error',
+      'vue/no-textarea-mustache': 'error',
+      'vue/no-unused-components': 'error',
+      'vue/no-unused-vars': 'error',
+      'vue/no-use-computed-property-like-method': 'error',
+      'vue/no-use-v-if-with-v-for': 'error',
+      'vue/no-useless-template-attributes': 'error',
+      'vue/no-v-for-template-key': 'error',
+      'vue/no-v-model-argument': 'error',
+      'vue/no-v-text-v-html-on-component': 'error',
+      'vue/require-component-is': 'error',
+      'vue/require-prop-type-constructor': 'error',
+      'vue/require-render-return': 'error',
+      'vue/require-v-for-key': 'error',
+      'vue/require-valid-default-prop': 'error',
+      'vue/return-in-computed-property': 'error',
+      'vue/return-in-emits-validator': 'error',
+      'vue/use-v-on-exact': 'error',
+      'vue/valid-attribute-name': 'error',
+      'vue/valid-define-emits': 'error',
+      'vue/valid-define-props': 'error',
+      'vue/valid-model-definition': 'error',
+      'vue/valid-next-tick': 'error',
+      'vue/valid-template-root': 'error',
+      'vue/valid-v-bind-sync': 'error',
+      'vue/valid-v-bind': 'error',
+      'vue/valid-v-cloak': 'error',
+      'vue/valid-v-else-if': 'error',
+      'vue/valid-v-else': 'error',
+      'vue/valid-v-for': 'error',
+      'vue/valid-v-html': 'error',
+      'vue/valid-v-if': 'error',
+      'vue/valid-v-model': 'error',
+      'vue/valid-v-on': 'error',
+      'vue/valid-v-once': 'error',
+      'vue/valid-v-pre': 'error',
+      'vue/valid-v-show': 'error',
+      'vue/valid-v-slot': 'error',
+      'vue/valid-v-text': 'error'
```

Comment:
If no `files` is specified, `(m|c)?js` will be used by default. 

question: is all these rules only applied to js files (no vue files)?

Reply:
I have added test cases and they work fine in `*.vue` as well.
Am I misunderstanding something?

https://github.com/vuejs/eslint-plugin-vue/pull/2436/commits/1438a8dfcb591df64e7a58503c7b959dc9b2fddd

Reply:
seems it's working fine with the preprocesser.